File: Job_applicator/test_wd.py
from playwright.sync_api import sync_playwright

def main():
    with sync_playwright() as p:
        browser = p.chromium.connect_over_cdp("http://localhost:9226")
        context = browser.contexts[0]
        for page in context.pages:
            if "tmobile.wd1" in page.url:
                print(f"URL: {page.url}")
                for btn in page.query_selector_all("button, a"):
                    txt = btn.inner_text().strip()
                    if txt: print(f" - {txt}")
                break

if __name__ == "__main__": main()

File: Job_applicator/test_wd.js
const { chromium } = require('playwright');
(async () => {
  const browser = await chromium.connectOverCDP('http://localhost:9226');
  const context = browser.contexts()[0];
  for (const page of context.pages()) {
    if (page.url().includes('tmobile.wd1')) {
      console.log('URL:', page.url());
      const locators = page.locator('button, a');
      const count = await locators.count();
      for (let i=0; i<count; i++) {
         const txt = await locators.nth(i).innerText().catch(()=>'');
         if (txt.trim()) console.log(' - ' + txt.trim().replace(/\n/g, ' '));
      }
      break;
    }
  }
  await browser.close();
})();
